js/modules/k6/data/share_test.go

```diff
@@ -141,14 +137,27 @@ func TestSharedArrayAnotherRuntimeExceptions(t *testing.T) {
 	for name, testCase := range cases {
 		name, testCase := name, testCase
 		t.Run(name, func(t *testing.T) {
-			_, err := rt.RunString(testCase.code)
+			t.Parallel()
+			testRuntime, err := newConfiguredRuntime(t)
+			rt := testRuntime.VU.Runtime()
+			require.NoError(t, err)
+			_, err = rt.RunString(makeArrayScript)
+			require.NoError(t, err)
+
+			testRuntime, err = configuredRuntimeFromAnother(t, testRuntime)
+			rt = testRuntime.VU.Runtime()
+			require.NoError(t, err)
+			_, err = rt.RunString(makeArrayScript)
+			require.NoError(t, err)
+			_, err = rt.RunString(testCase.code)
```

Comment:
The name of the Test suggests that the idea is that you will get this errors even from a different runtime using hte *same* sharedArray.

The previous code obviously failed on the *same* part. 

I am not certain if this was not due to some quirk of the previous implementation of SharedArray. As it might just be that that had a bug and this was trying to catch it. It might have even worked at some earlier version that had a global store for the sharedarrays :shrug:. 

I do find that modulestest having the capability to reuse the same modules for two different runtimes to be a thing that will be useful. There are likely to be more and more cases where modules will need to test that importing hte same module between VUs will have some specific behaviour and this makes it way easier to test .

Reply:
OK, clear, thanks for explaining.